CMakeLists.txt

```diff
@@ -380,241 +364,342 @@ if(${PREFIX}ENABLE_CODESIGN AND (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/cmake/codes
-				if (D_PLATFORM_BITS EQUAL 64)
-					set(_DOWNLOAD_OBS_HASH "SHA256=F15BC4CA8EB3F581A94372759CFE554E30D202B604B541445A5756B878E4E799")
-				else()
-					message(FATAL_ERROR "${LOGPREFIX} Platform '${D_PLATFORM_OS}' with architecture '${D_PLATFORM_ARCH}' and bitness '${D_PLATFORM_BITS}' is not supported.")
-					return()
+			elseif(D_PLATFORM_MAC)
+				if (D_PLATFORM_ARCH_X86)
+					if (D_PLATFORM_BITS EQUAL 64)
+						set(OBS_DOWNLOAD_URL "https://github.com/Xaymar/obs-studio/releases/download/27.2.4-ci/macos-x86_64.tar.gz")
+						set(OBS_DOWNLOAD_HASH "0DD5A57DD537B97CAA8470212F8F6B637F47D4742A5D1F17787F4FE9DBC70B33")
+					endif()
 				endif()
-			else()
-				message(FATAL_ERROR "${LOGPREFIX} Platform '${D_PLATFORM_OS}' with architecture '${D_PLATFORM_ARCH}' and bitness '${D_PLATFORM_BITS}' is not supported.")
+			endif()
+
+			# Verify that the platform, architecture and bitness is supported.
+			if(OBS_DOWNLOAD_URL STREQUAL "")
+				message(FATAL_ERROR "${LOGPREFIX} Download for libOBS failed, as Platform '${D_PLATFORM_OS}' with architecture '${D_PLATFORM_ARCH}' and bitness '${D_PLATFORM_BITS}' is not supported.")
 				return()
 			endif()
 		else()
-			message(FATAL_ERROR "${LOGPREFIX} Platform '${D_PLATFORM_OS}' with architecture '${D_PLATFORM_ARCH}' and bitness '${D_PLATFORM_BITS}' is not supported.")
-			return()
+			set(OBS_DOWNLOAD_URL "${${PREFIX}OBS_DOWNLOAD_URL}")
+			set(OBS_DOWNLOAD_HASH "${${PREFIX}OBS_DOWNLOAD_HASH}")
 		endif()
-	endif()
 
-	# Download libOBS
-	download_project(
-		PROJ libobs
-		URL "${_DOWNLOAD_OBS_URL}"
-		URL_HASH "${_DOWNLOAD_OBS_HASH}"
-		DOWNLOAD_NO_PROGRESS OFF
-		UPDATE_DISCONNECTED OFF
-	)
+		if(OBS_DOWNLOAD_HASH STREQUAL "")
+			download_project(
+				PROJ libobs
+				PREFIX "autodeps"
+				URL "${OBS_DOWNLOAD_URL}"
+				DOWNLOAD_NO_PROGRESS OFF
+				UPDATE_DISCONNECTED OFF
+			)
+		else()
+			download_project(
+				PROJ libobs
+				PREFIX "autodeps"
+				URL "${OBS_DOWNLOAD_URL}"
+				URL_HASH "SHA256=${OBS_DOWNLOAD_HASH}"
+				DOWNLOAD_NO_PROGRESS OFF
+				UPDATE_DISCONNECTED OFF
+			)
+		endif()
+		CacheSet(${PREFIX}OBS_PATH "${libobs_SOURCE_DIR}")
+	endif()
+	
+	if(NOT EXISTS "${${PREFIX}OBS_PATH}/cmake/LibObs/LibObsConfig.cmake")
+		message(FATAL_ERROR "${LOGPREFIX} The provided path for libOBS is invalid as it did not contain '/cmake/LibObs/LibObsConfig.cmake'.")
+		return()
+	else()
+		include("${${PREFIX}OBS_PATH}/cmake/LibObs/LibObsConfig.cmake")
+	endif()
 
-	include("${libobs_SOURCE_DIR}/cmake/LibObs/LibObsConfig.cmake")
+	if (NOT TARGET obs-frontend-api)
+		if(EXISTS "${${PREFIX}OBS_PATH}/cmake/obs-frontend-api/obs-frontend-apiConfig.cmake")
+			include("${${PREFIX}OBS_PATH}/cmake/obs-frontend-api/obs-frontend-apiConfig.cmake")
+		elseif((EXISTS "${${PREFIX}OBS_PATH}/bin/${D_PLATFORM_BITS}bit/obs-frontend-api.lib") AND (EXISTS "${${PREFIX}OBS_PATH}/include/obs-frontend-api.h"))
```

Comment:
What is the `obs-frontend-api.lib` file? If it's the obs-frontend-api shared library, its path for Linux should be: `${${PREFIX}OBS_PATH}/lib/libobs-frontend-api.so`

Reply:
The .lib file is the linker file for obs-frontend-api.so/.dll. It should be present in your build.

Reply:
Nevermind, it directly links to the .so file instead. The .lib file is only with some compilers and flags.

Reply:
Hmm how are configurations that have the `libobs-frontend-api.so` and the `obs-frontend-api.h` files but not the `obs-frontend-apiConfig.cmake` or the `obs-frontend-api.lib` files getting the UI?

Reply:
All my builds have the ...Config.cmake file, as well as a .lib file. That OBS still doesn't export a .pc or .cmake file for the frontend-api is probably a bug.

Reply:
From what I read, the `.lib` files are Windows specific.

I've opened https://github.com/obsproject/obs-studio/issues/6556 for the CMake files.

Reply:
It depends on the compiler/linker used. I'll adjust it to check for .so on unix, just to be on the safer side. Generally these cmake files should be present, and any project not exporting them (as well as a pc file for pkg-config) in a package step is technically broken or not meant to have plugins.